File: Valid_Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        valid = False
        stack = []
        if len(s)%2 == 0:
            for elements in range(len(s)):
                if s[elements] in ['(', '[', '{']:
                    stack.append(s[elements])

                elif (stack and stack[-1] == '(' and s[elements] == ')' ):
                        stack.pop(-1)
                        valid = True
                elif (stack and stack[-1] == '[' and s[elements] == ']'):
                        print(stack)
                        stack.pop(-1)
                        valid = True   
                elif (stack and stack[-1]  == '{' and s[elements] == '}'):
                        
                        stack.pop(-1)
                        valid = True 
                else:
                        valid = False            
            if stack:
                valid = False            


                                  
        return valid            

File: Valid_Parentheses.c
bool isValid(char * s)
{
    bool value = false;
    for(int i=0; s[i]!='\0'; i++)
    {
        switch(s[i])
        {
            case '(':
                if(s[i+1] == ')' && s[i+1]!='\0')
                {
                    value = true;
                    i++;
                }
                else
                {
                    value = false;
                }
            break;

            case '[':
                if(s[i+1] == ']' && s[i+1]!='\0')
                {
                    value = true;
                    i++;
                }
                else
                {
                    value = false;
                }
            break;
            case '{':
                if(s[i+1] == '}' && s[i+1]!='\0')
                {

                    value = true;
                    i++;
                }
                
                else
                {
                    value = false;
                }
            break;
        }
    }

return value;
}